### PR TITLE
[LM-9615] Added CustomizableUI

### PR DESCRIPTION
**Pull request by prabha-likeminds**
## Description

**Features**
- Added customizable UI

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [ ] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [ ] Label added
- [ ] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [ ] Description about the Feature/ Bugs added

### CODE REVIEW

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 1]
```diff
@@ -57,7 +77,7 @@ const UniversalFeed = () => {
   async function fetchFeed() {
     let payload = {
       page: feedPageNumber,
-      pageSize: 10,
+      pageSize: 30,
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:80]: pageSize shouldn't more than 20

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 2]
```diff
@@ -109,23 +129,83 @@ const UniversalFeed = () => {
...
+
+  // this function handles the functionality on the pin option
+  const handlePinPost = async (id: string) => {
+    let payload = {
+      postId: id,
+    };
+    dispatch(pinPostStateHandler(payload.postId) as any);
+    let pinPostResponse = await dispatch(
+      pinPost(
+        PinPostRequest.builder().setpostId(payload.postId).build(),
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:152]: move `setpostId()` and `build()` to next line to maintain readability of the code.

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 3]
```diff
@@ -109,23 +129,83 @@ const UniversalFeed = () => {
...
+      handleReportPost();
+    }
+    if (itemId === DELETE_POST_MENU_ITEM) {
+      handleDeletePost(true);
+    }
+  };
+
+  const getPostDetail = () => {
+    const postDetail = feedData.find(
+      (item: any) => item.Id === selectedMenuItemPostId,
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:184]: why `any`?

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 4]
```diff
@@ -109,23 +129,83 @@ const UniversalFeed = () => {
...
+        } else {
+          // shared
+        }
+      } else if (result.action === Share.dismissedAction) {
+        // dismissed
+      }
+    } catch (error: any) {
+      Alert.alert(error.message);
+    }
+  };
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:207]: move this code to a new file as `ShareUtils` as Share function will be total customer dependent and it should be properly readable.

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 5]
```diff
@@ -134,11 +214,45 @@ const UniversalFeed = () => {
           data={feedData}
           renderItem={({item}: any) => (
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:215]: why `any`?

File: FeedSX/screens/UniversalFeed/index.tsx [hunk 6]
```diff
@@ -134,11 +214,45 @@ const UniversalFeed = () => {
...
+                    savePostHandler(item.Id);
+                  },
+                },
+                likeTextButton: {
+                  onTap: () => NavigationService.navigate('LikesList', item.Id),
+                },
+              }}
             />
           )}
           estimatedItemSize={200}
```
Comment (ishaan1607) [inline FeedSX/screens/UniversalFeed/index.tsx:258]: why it is `200`?

File: FeedSX/viewDataModels/postViewData/index.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+export function postViewData(data: any): LMPostUI[] {
```
Comment (ishaan1607) [inline FeedSX/viewDataModels/postViewData/index.tsx:1]: rename function name to `convertToLMPostUI()`

Comment (ishaan1607) [inline FeedSX/viewDataModels/postViewData/index.tsx:1]: Also, why input parameter is `any` , it should be the interface/class returned from data layer

File: FeedSX/viewDataModels/postViewData/index.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+      isSaved: item.isSaved,
+      likesCount: item.likesCount,
+      menuItems: item.menuItems,
+      tempId: item.tempId,
+      text: item.text,
+      updatedAt: item.updatedAt,
+      userId: item.userId,
+      uuid: item.uuid,
+      user: userData[item.userId]
+    };
```
Comment (ishaan1607) [inline FeedSX/viewDataModels/postViewData/index.tsx:24]: where are the converters for `LMAttachmentUI` , `LMAttachmentMetaUI` and `LMUserUI`

File: FeedSX/viewDataModels/postViewData/index.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
+export function postViewData(data: any): LMPostUI[] {
+  let postData = data?.posts;
+  let userData = data?.users;
+  return postData.map((item: any) => {
+    return {
```
Comment (ishaan1607) [inline FeedSX/viewDataModels/postViewData/index.tsx:5]: we should mention the name of the interface exported like here it should be `LMPostUI`

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: FeedSX/screens/UniversalFeed/index.tsx (hunk 1 above)
Comment (prabha-likeminds) [reply FeedSX/screens/UniversalFeed/index.tsx:80]: resolved

File: FeedSX/screens/UniversalFeed/index.tsx (hunk 2 above)
Comment (prabha-likeminds) [reply FeedSX/screens/UniversalFeed/index.tsx:152]: resolved

**Review by ishaan1607 — APPROVED** [review]

**Pull request merged**